apps/web/src/components/Feed/Socialize/CommentsModal/CommentsModal.tsx

```diff
@@ -99,6 +133,78 @@ export function CommentsModal({
     loadPrevious(NOTES_PER_PAGE);
   }, [loadPrevious]);
 
+  const rowRepliesExpanded = useState<Record<number, boolean>>({});
+
+  const setRowRepliesExpanded = useCallback(
+    (index: number, value: boolean) => {
+      rowRepliesExpanded[1]((prev) => {
+        return {
+          ...prev,
+          [index]: value,
+        };
+      });
+    },
+    [rowRepliesExpanded]
+  );
+
+  const getRowRepliesExpanded = useCallback(
+    (index: number) => {
+      return rowRepliesExpanded[0][index] ?? false;
+    },
+    [rowRepliesExpanded]
+  );
+
+  const [contentHeight, setContentHeight] = useState(0);
+
+  const isRowLoaded = ({ index }: { index: number }) => !hasPrevious || index < comments.length;
+
+  const rowCount = hasPrevious ? comments.length + 1 : comments.length;
+
+  const calculateModalMaxHeight = useCallback(() => {
+    let height = 0;
+    for (let i = 0; i < rowCount; i++) {
+      height += measurerCache.rowHeight({ index: i });
+    }
+    const modalMaxHeight = fullscreen ? window.innerHeight : 640;
+    // 121 is the height of the modal header + bottom padding + comment box
+    setContentHeight(Math.min(height, modalMaxHeight - 121));
+  }, [fullscreen, rowCount, measurerCache, setContentHeight]);
+
+  const recalculateHeightsWhenCommentsChange = useCallback(() => {
+    measurerCache.clearAll();
+    virtualizedListRef.current?.recomputeRowHeights();
+    calculateModalMaxHeight();
+  }, [calculateModalMaxHeight, measurerCache]);
+
+  const handleSubmitComment = useCallback(
+    async (comment: string, mentions: MentionInput[]) => {
+      await onSubmitComment(
+        comment,
+        mentions,
+        selectedComment?.commentId,
+        topCommentId?.current ?? ''
+      );
+      setSelectedComment(null);
+      topCommentId.current = null;
+      recalculateHeightsWhenCommentsChange();
+    },
+    [onSubmitComment, selectedComment?.commentId, recalculateHeightsWhenCommentsChange]
+  );
+
+  useEffect(recalculateHeightsWhenCommentsChange, [
+    comments,
+    measurerCache,
+    recalculateHeightsWhenCommentsChange,
+  ]);
+
+  // calculate the height of the list
+  useEffect(() => {
+    calculateModalMaxHeight();
+    //
+    // limit dependencies. we specifically want to run this effect only when the rowHeightCache changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [measurerCache._rowHeightCache]);
```

Comment:
hmm i wonder if having both of these effects is overkill 🤔 

the second effect calls `calculateModalMaxHeight`, but that function is also called by `recalculateHeightsWhenCommentsChange` – so we might be able to get rid of the second effect. maybe let's just toss it and play around / see if it breaks

Reply:
I tried remove the second `useEffects` and seems like it broke the modal 😅. I tried to merge the effects into one and it still broken lol



![CleanShot 2023-12-04 at 08 52 33@2x](https://github.com/gallery-so/gallery/assets/4480258/9eb983ca-73ac-444c-bc7f-562d357ed68f)